java/integrated-windows-authentication/src/main/java/com/example/arcgisruntime/sample/integratedwindowsauthentication/MainActivity.java

```diff
@@ -0,0 +1,271 @@
+
+        // search the portal for web maps
+        ListenableFuture<PortalQueryResultSet<PortalItem>> portalItemResult = portal
+            .findItemsAsync(new PortalQueryParameters("type:(\"web map\" NOT \"web mapping application\")"));
+        portalItemResult.addDoneListener(() -> {
+          try {
+            PortalQueryResultSet<PortalItem> portalItemSet = portalItemResult.get();
+            PortalItemAdapter portalItemAdapter = new PortalItemAdapter(portalItemSet.getResults(),
+                portalItem -> addMap(portal, portalItem.getItemId()));
+            mRecyclerView.setAdapter(portalItemAdapter);
+            mPortalLoadStateView.setVisibility(View.GONE);
+          } catch (ExecutionException | InterruptedException e) {
+            String error = "Error getting portal item set from portal: " + e.getMessage();
+            Toast.makeText(this, error, Toast.LENGTH_LONG).show();
+            Log.e(TAG, error);
+          }
+        });
+      } else {
+        ArcGISRuntimeException loadError = portal.getLoadError();
+        String error = loadError.getErrorCode() == 17 ?
+            "Portal sign in was cancelled by user." :
+            "Portal sign in failed: " + portal.getLoadError().getCause().getMessage();
+        Toast.makeText(this, error, Toast.LENGTH_LONG).show();
+        Log.e(TAG, error);
+      }
+    });
+  }
+
+  /**
+   * Add the given portal item to a new map and set the map to the map view.
+   *
+   * @param portal
+   * @param itemId
+   */
+  private void addMap(Portal portal, String itemId) {
+    // report error and return if portal is null
+    if (portal == null) {
+      String error = "Portal not instantiated.";
+      Toast.makeText(this, error, Toast.LENGTH_LONG).show();
+      Log.e(TAG, error);
+      return;
+    }
+    // use the item ID to create a portal item from the portal
+    PortalItem portalItem = new PortalItem(portal, itemId);
+    // create a map using the web map (portal item) and add it to the map view
+    ArcGISMap webMap = new ArcGISMap(portalItem);
+    mMapView.setMap(webMap);
+    // show item ID in UI
+    mLoadWebMapTextView.setText("Loaded web map from item " + itemId);
+  }
+
+  /**
+   * When a user credential challenge is issued, a dialog will be presented to the user to take credential information.
+   * The portal URL will be displayed as a message in the dialog. If a wrong credential has been passed in the previous
+   * attempt, a different message will be displayed in the dialog. The dialog has two edit text boxes for username and
+   * password respectively. Other SDKs' samples may have one more parameter for IWA domain. As indicated by the Javadoc
+   * of UseCredential the SDK is in favor of passing username as username@domain.
+   */
+  @Override public AuthenticationChallengeResponse handleChallenge(AuthenticationChallenge authenticationChallenge) {
+    if (authenticationChallenge.getType() == AuthenticationChallenge.Type.USER_CREDENTIAL_CHALLENGE) {
```

Comment:
Some public web maps coming from "Search public" such as "World Demographics" require ArcGIS Online subscription to load some layers, which causes our SDK issue a challenge with error saying "Token required". As a result this sample will prompt for credentials when loading these layers. As this sample showcase IWA authentication, I think we should let those layers fail to load silently instead of prompting for credentials. if so, we should change line 188 to:

```java
if ((authenticationChallenge.getType() == AuthenticationChallenge.Type.USER_CREDENTIAL_CHALLENGE)
    && (authenticationChallenge.getRemoteResource() instanceof Portal)) {
```

Reply:
Good catch and I think you're right about letting it fail silently.